src/main/java/it/gov/pagopa/pu/debtpositions/service/create/debtposition/CreateDebtPositionServiceImpl.java

```diff
@@ -0,0 +1,70 @@
+import it.gov.pagopa.pu.debtpositions.exception.custom.ConflictErrorException;
+import it.gov.pagopa.pu.debtpositions.repository.InstallmentNoPIIRepository;
+import it.gov.pagopa.pu.debtpositions.service.AuthorizeOperatorOnDebtPositionTypeService;
+import it.gov.pagopa.pu.debtpositions.service.DebtPositionService;
+import it.gov.pagopa.pu.debtpositions.service.create.GenerateIuvService;
+import it.gov.pagopa.pu.debtpositions.service.create.ValidateDebtPositionService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class CreateDebtPositionServiceImpl implements CreateDebtPositionService {
+
+  private final AuthorizeOperatorOnDebtPositionTypeService authorizeOperatorOnDebtPositionTypeService;
+  private final ValidateDebtPositionService validateDebtPositionService;
+  private final DebtPositionService debtPositionService;
+  private final GenerateIuvService generateIuvService;
+  private final InstallmentNoPIIRepository installmentNoPIIRepository;
+
+  public CreateDebtPositionServiceImpl(AuthorizeOperatorOnDebtPositionTypeService authorizeOperatorOnDebtPositionTypeService,
+                                       ValidateDebtPositionService validateDebtPositionService,
+                                       DebtPositionService debtPositionService, GenerateIuvService generateIuvService,
+                                       InstallmentNoPIIRepository installmentNoPIIRepository) {
+    this.authorizeOperatorOnDebtPositionTypeService = authorizeOperatorOnDebtPositionTypeService;
+    this.validateDebtPositionService = validateDebtPositionService;
+    this.debtPositionService = debtPositionService;
+    this.generateIuvService = generateIuvService;
+    this.installmentNoPIIRepository = installmentNoPIIRepository;
+  }
+
+  @Override
+  public DebtPositionDTO createDebtPosition(DebtPositionDTO debtPositionDTO, Boolean massive, Boolean pagopaPayment, String accessToken, String operatorExternalUserId) {
+    log.info("Creating a DebtPosition having organizationId {}, debtPositionTypeOrgId {}, iupdOrg {}, ingestionFlowFileId {}, rowNumber {}", debtPositionDTO.getOrganizationId(),
+      debtPositionDTO.getDebtPositionTypeOrgId(), debtPositionDTO.getIupdOrg(), debtPositionDTO.getIngestionFlowFileId(), debtPositionDTO.getIngestionFlowFileLineNumber());
+
+    authorizeOperatorOnDebtPositionTypeService.authorize(debtPositionDTO.getOrganizationId(), debtPositionDTO.getDebtPositionTypeOrgId(), operatorExternalUserId);
+    validateDebtPositionService.validate(debtPositionDTO, accessToken);
+    verifyInstallmentUniqueness(debtPositionDTO);
+
+    if (Boolean.TRUE.equals(pagopaPayment)) {
+      debtPositionDTO.getPaymentOptions().stream()
+        .flatMap(po -> po.getInstallments().stream())
+        .forEach(installment -> {
+          Long orgId = debtPositionDTO.getOrganizationId();
+          String generatedIuv = generateIuvService.generateIuv(String.valueOf(orgId), accessToken);
```

Comment:
dont' transform the organizationId in String here
let's use it as Long until the OrganizationSearchClient (here you will invoke String.valueOf), the fact that API is asking it as String is something due to the REST integration, let's hide this detail until that class, using Long instead